packages/contracts/evm-contracts/contracts/orderbook/OrderbookDex.sol

```diff
@@ -3,41 +3,59 @@
 pragma solidity ^0.8.20;
 
 import {Address} from "@openzeppelin/contracts/utils/Address.sol";
-import {ERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+import {ERC1155Holder} from "@openzeppelin/contracts/token/ERC1155/utils/ERC1155Holder.sol";
 import {IERC165} from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
 import {ReentrancyGuard} from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+
+import {IInverseProjected1155} from "../token/IInverseProjected1155.sol";
 import {IOrderbookDex} from "./IOrderbookDex.sol";
 
 /// @notice Facilitates trading an asset that is living on a different app-chain.
-contract OrderbookDex is IOrderbookDex, ERC165, ReentrancyGuard {
+contract OrderbookDex is IOrderbookDex, ERC1155Holder, ReentrancyGuard {
     using Address for address payable;
 
-    mapping(address => mapping(uint256 => Order)) public orders;
-    mapping(address => uint256) public sellersOrderId;
+    IInverseProjected1155 internal asset;
+    mapping(address => mapping(uint256 => Order)) internal orders;
+    mapping(address => uint256) internal sellersOrderId;
 
     error OrderDoesNotExist(uint256 orderId);
     error InsufficientEndAmount(uint256 expectedAmount, uint256 actualAmount);
     error InvalidInput(uint256 input);
     error InvalidInputArity();
 
+    constructor(IInverseProjected1155 _asset) {
+        asset = _asset;
+    }
+
+    /// @notice Returns the address of the asset that is being traded.
+    function getAsset() public view virtual returns (address) {
+        return address(asset);
+    }
+
     /// @notice Returns the seller's current `orderId` (index that their new sell order will be mapped to).
-    function getSellerOrderId(address seller) external view returns (uint256) {
+    function getSellerOrderId(address seller) public view virtual returns (uint256) {
         return sellersOrderId[seller];
     }
 
     /// @notice Returns the Order struct information about an order identified by the combination `<seller, orderId>`.
-    function getOrder(address seller, uint256 orderId) external view returns (Order memory) {
+    function getOrder(address seller, uint256 orderId) public view virtual returns (Order memory) {
         return orders[seller][orderId];
     }
 
     /// @notice Creates a sell order with incremental seller-specific `orderId` for the specified `assetAmount` at specified `pricePerAsset`.
     /// @dev The order information is saved in a nested mapping `seller address -> orderId -> Order`.
     /// MUST emit `OrderCreated` event.
-    function createSellOrder(uint256 assetAmount, uint256 pricePerAsset) public {
+    function createSellOrder(
```

Comment:
we probably want a batch version of this function like `onERC1155BatchReceived` because probably people will want to make a single transaction that creates all the sell orders for all the tokens they have instead of doing it one assetId at a time

Reply:
Added.